addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -300,8 +305,10 @@ func (wh *ClusterBootstrap) periodicGVRCachesClean(ctx context.Context) {
 			ticker.Stop()
 			return
 		case <-ticker.C:
+			wh.lock.Lock()
 			wh.cachedDiscoveryClient.Invalidate()
 			wh.providerGVR = make(map[schema.GroupKind]*schema.GroupVersionResource)
+			wh.lock.Unlock()
```

Comment:
Just to be extra safe, I would wrap this in a function and `defer` unlock mutex:

```go
func () {
    wh.lock.Lock()
    defer wh.lock.Unlock()
    wh.cachedDiscoveryClient.Invalidate()
    wh.providerGVR = make(map[schema.GroupKind]*schema.GroupVersionResource)
}()
```

Reply:
Good suggestion, had to merge same patch in release-0.25 branch for wcp code drop yesterday so gonna keep the same on main for consistency. Also I looked at Invalidate()(where a potential panic can happen) and it is minimal(below) so the potential chance of panic is probably zero before Unlock() can run. 


```
func (d *memCacheClient) Invalidate() {
	d.lock.Lock()
	defer d.lock.Unlock()
	d.cacheValid = false
	d.groupToServerResources = nil
	d.groupList = nil
}
```